### PR TITLE
Fix jaas demos

### PR DESCRIPTION
**Pull request by janbartel**
There can only be a single jaas configuration file, so changed the demos to that instead of multiple ones (which were being ignored anyway).

### CODE REVIEW

File: jetty-ee10/jetty-ee10-demos/jetty-ee10-demo-jaas-webapp/src/main/config/modules/ee10-demo-jaas.mod [hunk 1]
```diff
@@ -20,7 +20,6 @@ ext
 
 [files]
 basehome:modules/demo.d/ee10-demo-jaas.xml|webapps/ee10-demo-jaas.xml
-basehome:modules/demo.d/ee10-demo-login.conf|etc/ee10-demo-login.conf
```
Comment (sbordet) [inline jetty-ee10/jetty-ee10-demos/jetty-ee10-demo-jaas-webapp/src/main/config/modules/ee10-demo-jaas.mod:23]: Why this? 

All the config files necessary were in the `demo.d` directory and copied to relevant places.
Why is now `ee10-demo-login.conf` an exception?

File: jetty-ee8/jetty-ee8-demos/jetty-ee8-demo-jaas-webapp/pom.xml [hunk 2]
```diff
@@ -26,14 +26,14 @@
               <!-- This is for convenience so that the src/etc/login.conf file can stay unmodified when copied to $jetty.home/etc directory -->
                 <jetty.base>${basedir}/src/main/config/modules/demo.d</jetty.base>
               <!-- Mandatory. This system property tells JAAS where to find the login module configuration file -->
-                <java.security.auth.login.config>${basedir}/src/main/config/modules/demo.d/ee8-demo-login.conf</java.security.auth.login.config>
```
Comment (sbordet) [inline jetty-ee8/jetty-ee8-demos/jetty-ee8-demo-jaas-webapp/pom.xml:29]: I understand ee8 demos are derived from ee9 demos with some Maven magic.
As such we should just change the ee9 files, not the ee8 ones?

File: jetty-home/src/main/resources/modules/demo-jaas.mod [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+[tags]
+demo
+webapp
+
+[depends]
+core-deploy
+
+[files]
+basehome:modules/demo.d/demo-login.conf|etc/demo-login.conf
```
Comment (sbordet) [inline jetty-home/src/main/resources/modules/demo-jaas.mod:13]: This line was removed from the eeX versions?

File: jetty-home/src/main/resources/modules/demo.d/demo-login.conf [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+org.eclipse.jetty.ee9.jaas.spi.PropertyFileLoginModule required
+debug="true"
+file="${jetty.base}/etc/ee9-demo-login.properties";
+};
+
+ee8-xyz {
+org.eclipse.jetty.ee8.jaas.spi.PropertyFileLoginModule required
+debug="true"
+file="${jetty.base}/etc/ee8-demo-login.properties";
+};
```
Comment (sbordet) [inline jetty-home/src/main/resources/modules/demo.d/demo-login.conf:17]: Why is a core demo having eeX configuration?

**Review by sbordet — CHANGES_REQUESTED** [review]

File: jetty-ee10/jetty-ee10-demos/jetty-ee10-demo-jaas-webapp/src/main/config/modules/ee10-demo-jaas.mod (hunk 1 above)
Comment (sbordet) [reply jetty-ee10/jetty-ee10-demos/jetty-ee10-demo-jaas-webapp/src/main/config/modules/ee10-demo-jaas.mod:23]: Also, the `ee10-demo-login-conf` file still references module `xyz` but should be `ee10-xyz`.

**Review by sbordet — CHANGES_REQUESTED** [review]

File: jetty-ee8/jetty-ee8-demos/jetty-ee8-demo-jaas-webapp/pom.xml (hunk 2 above)
Comment (olamy) [reply jetty-ee8/jetty-ee8-demos/jetty-ee8-demo-jaas-webapp/pom.xml:29]: demos cannot be derived from ee9 demos. 
they are including some very differents jars etc... 
so I couldn;t do it for demos.

**Comment by sbordet** [comment]
@janbartel I'm not sure I follow these changes.

Build fails due to circular dependency.

If I want to just deploy the ee10 jaas demo, does it work?

And why the core demo has a configuration referring to eeX?

Are there distribution tests for these demos?

File: jetty-ee10/jetty-ee10-demos/jetty-ee10-demo-jaas-webapp/src/main/config/modules/ee10-demo-jaas.mod (hunk 1 above)
Comment (janbartel) [reply jetty-ee10/jetty-ee10-demos/jetty-ee10-demo-jaas-webapp/src/main/config/modules/ee10-demo-jaas.mod:23]: I thought I made the problem clean in the opening comment of the PR? The problem is that you cannot have more than one jaas config file, so all of the eeX-demo-login.conf files were being ignored. All of the jaas configuration must go into a single file.

File: jetty-home/src/main/resources/modules/demo-jaas.mod (hunk 3 above)
Comment (janbartel) [reply jetty-home/src/main/resources/modules/demo-jaas.mod:13]: Exactly. As there can only be a single config file.

File: jetty-home/src/main/resources/modules/demo.d/demo-login.conf (hunk 4 above)
Comment (janbartel) [reply jetty-home/src/main/resources/modules/demo.d/demo-login.conf:17]: If you look at  `jetty-home/src/main/resources/modules/demo.d/` you will see lots of other eeX related config files.

**Comment by janbartel** [comment]
@sbordet  as I've said, the problem I'm trying to fix is that you cannot have multiple jaas config files. Thus you cannot have one for ee8, one for ee9 and one for ee10. All config must be in a single file.

As for deploying a single webapp, yes I need to do a bit more work on that.

The circular dependency is this:

> Caused by: org.codehaus.plexus.util.dag.CycleDetectedException: Edge between 'Vertex{label='org.eclipse.jetty:jetty-home:12.0.0-SNAPSHOT'}' and 'Vertex{label='org.eclipse.jetty.ee10:jetty-ee10-home:12.0.0-SNAPSHOT'}' introduces to cycle in the graph org.eclipse.jetty.ee10:jetty-ee10-home:12.0.0-SNAPSHOT --> org.eclipse.jetty.ee10.demos:jetty-ee10-demo-jaas-webapp:12.0.0-SNAPSHOT --> org.eclipse.jetty.ee10:jetty-ee10-maven-plugin:12.0.0-SNAPSHOT --> org.eclipse.jetty:jetty-home:12.0.0-SNAPSHOT --> org.eclipse.jetty.ee10:jetty-ee10-home:12.0.0-SNAPSHOT

Having fixed the coordinates of the `jetty-maven-plugin` in the `jetty-ee10-demo-jaas-webapp` project, there is now a circular reference around `jetty-home` and the `jetty-maven-plugin` via the webapp. We seem to have lost the ability to be able to use the plugin for our webapp projects?!

**Comment by olamy** [comment]
> 
> No idea why there is now a circular dependency having just changed some `.mod` files, will look at that.

you changed a pom as well ;) and that's probably the reason.
probably something such:
maven-plugin need jetty-home, jetty-home need jaas-demo but jaas-demo needs  maven-plugin

**Comment by sbordet** [comment]
@janbartel I still don't understand.
If I only deploy the ee10 jaas demo, there will be a single file `ee10-demo-login.conf`, no?
That "there must be only one jaas config file" is already true if I only deploy one eeX jaas demo, whatever it is.

Are you trying to make the ee-10-demo module depend on the equivalent core module just in case someone deploys multiple eeX jaas demos?

**Comment by janbartel** [comment]
@olamy but `jetty-maven-plugin` _doesn't_ need `jetty-home`.  Only the _integration tests_ for the jetty-maven-plugin need jetty-home. If they were moved out of jetty-maven-plugin then there wouldn't be a problem.

**Comment by janbartel** [comment]
@sbordet you need to be able to deploy `demos` and have it work. Currently if you do that, then all of the jaas demos are broken. As you point out, you also need to be able to deploy just a single `eeX-demo` as well.

**Comment by olamy** [comment]
> @olamy but `jetty-maven-plugin` _doesn't_ need `jetty-home`. Only the _integration tests_ for the jetty-maven-plugin need jetty-home. If they were moved out of jetty-maven-plugin then there wouldn't be a problem.

which is **real** pain when developing/debugging it... :(  as you have to build multiple modules first
I'm not sure why adding jetty-eeX-maven-plugin to jaas-demo.
As long as you have distribution tests covering jaas-eeX-demo it's fine

**Comment by janbartel** [comment]
> > @olamy but `jetty-maven-plugin` _doesn't_ need `jetty-home`. Only the _integration tests_ for the jetty-maven-plugin need jetty-home. If they were moved out of jetty-maven-plugin then there wouldn't be a problem.
> 
> which is **real** pain when developing/debugging it... :( as you have to build multiple modules first I'm not sure why adding jetty-eeX-maven-plugin to jaas-demo. As long as you have distribution tests covering jaas-eeX-demo it's fine

@olamy I'm not _adding_ the maven plugin to the demo webapps, it was always there ;) and it used to work. I think it's a reasonable expectation that you could run our demo webapps just like you could run any other webapp with jetty.

**Comment by olamy** [comment]
> > > @olamy but `jetty-maven-plugin` _doesn't_ need `jetty-home`. Only the _integration tests_ for the jetty-maven-plugin need jetty-home. If they were moved out of jetty-maven-plugin then there wouldn't be a problem.
> > 
> > 
> > which is **real** pain when developing/debugging it... :( as you have to build multiple modules first I'm not sure why adding jetty-eeX-maven-plugin to jaas-demo. As long as you have distribution tests covering jaas-eeX-demo it's fine
> 
> @olamy I'm not _adding_ the maven plugin to the demo webapps, it was always there ;) and it used to work. I think it's a reasonable expectation that you could run our demo webapps just like you could run any other webapp with jetty.

oh right. correct that;s weird that should fail as well in `jetty-10.0.x` branch. 
We have been lucky :P.Sounds like a maven bug as it's similar as `jetty-10.0.x` branch except here there is one more module in the stack (e.g `jetty-ee10-home`)

**Review by olamy — APPROVED** [review]

**Pull request merged**